codebasin/config.py

```diff
@@ -247,25 +247,36 @@ def load_codebase(config, rootdir):
     cfg_codebase = config["codebase"]
     if not cfg_codebase:
         raise RuntimeError("Empty 'codebase' section found in config file!")
-    if "files" not in cfg_codebase or cfg_codebase["files"] == []:
-        raise RuntimeError("Empty 'files' section found in codebase definition!")
+    if "files" not in cfg_codebase:
+        raise RuntimeError("No 'files' section found in codebase definition!")
     if "platforms" not in cfg_codebase or cfg_codebase["platforms"] == []:
         raise RuntimeError("Empty 'platforms' section found in codebase definition!")
 
-    codebase = {"files": list(it.chain(*(expand_path(os.path.join(rootdir, f))
-                                         for f in cfg_codebase["files"]))),
-                "platforms": cfg_codebase["platforms"]}
+    codebase = {}
+
+    codebase["platforms"] = cfg_codebase["platforms"]
 
     if "exclude_files" in cfg_codebase:
         codebase["exclude_files"] = frozenset(it.chain(*(expand_path(os.path.join(rootdir, f))
                                                          for f in cfg_codebase["exclude_files"])))
     else:
         codebase["exclude_files"] = frozenset([])
 
-    # Ensure that the codebase actually specifies valid files
-    if not codebase["files"]:
-        raise RuntimeError("Specified codebase contains no valid files -- " +
-                           "regular expressions and/or working directory may be incorrect.")
+    if cfg_codebase["files"]:
+        codebase["files"] = list(it.chain(*(expand_path(os.path.join(rootdir, f))
+                                            for f in cfg_codebase["files"])))
+        if not codebase["files"]:
+            raise RuntimeError("Codebase configuration contains no valid files. " +
```

Comment:
Not important at all, but you don't actually need `+` to concatenate strings

Reply:
I didn't know that!  I'm going to merge as is, with hopes that the big formatting pass will clean this up.  If it doesn't, I'll manually go through and look for all instances of this and fix them manually.